components/supervisor/frontend/src/index.ts

```diff
@@ -10,9 +10,37 @@
  */
 import { IDEMetricsServiceClient, MetricsName } from "./ide/ide-metrics-service-client";
 IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendClientTotal).catch(() => {})
-window.addEventListener('error', () => {
-    IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendErrorTotal).catch(() => {})
-})
+
+declare global {
+    interface Window {
+        gitpodIDEErrors: Map<string, string>[];
```

Comment:
we will need to make sure that both stable and latest have these changes at the time of supervisor deployment or it will break